src/Instrumentation/ExtRdKafka/src/ExtRdKafkaInstrumentation.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Contrib\Instrumentation\ExtRdKafka;
+
+use Composer\InstalledVersions;
+use OpenTelemetry\API\Globals;
+use OpenTelemetry\API\Instrumentation\CachedInstrumentation;
+use OpenTelemetry\API\Trace\Span;
+use OpenTelemetry\API\Trace\SpanKind;
+use OpenTelemetry\Context\Context;
+use function OpenTelemetry\Instrumentation\hook;
+use OpenTelemetry\SemConv\TraceAttributes;
+
+use OpenTelemetry\SemConv\TraceAttributeValues;
+
+use RdKafka\KafkaConsumer;
+use RdKafka\Message;
+use RdKafka\ProducerTopic;
+
+use Throwable;
+
+class ExtRdKafkaInstrumentation
+{
+    public const NAME = 'ext_rdkafka';
+    public static function register(): void
+    {
+        $instrumentation = new CachedInstrumentation(
+            'io.opentelemetry.contrib.php.ext_rdkafka',
+            InstalledVersions::getVersion('open-telemetry/opentelemetry-auto-ext-rdkafka'),
+            'https://opentelemetry.io/schemas/1.25.0',
+        );
+
+        // Start root span and propagate parent if it exists in headers, for each message consumed
+        self::addConsumeHooks($instrumentation);
+        // End root span on offset commit
+        self::addCommitHooks('commit');
+        self::addCommitHooks('commitAsync');
+        // Context propagation for outbound messages
+        self::addProductionHooks($instrumentation);
+    }
+
+    private static function addCommitHooks($functionName)
+    {
+        hook(
+            KafkaConsumer::class,
+            $functionName,
+            null,
+            static function () {
+                $scope = Context::storage()->scope();
+                if ($scope === null) {
+                    return;
+                }
+                $scope->detach();
+                $span = Span::fromContext($scope->context());
+                $span->end();
+            }
+        );
+    }
+
+    private static function addProductionHooks($instrumentation)
+    {
+        hook(
+            ProducerTopic::class,
+            'producev',
+            static function (
+                ProducerTopic $exchange,
+                array $params,
+                string $class,
+                string $function,
+                ?string $filename,
+                ?int $lineno
+            ) use ($instrumentation) : array {
+                /** @var CachedInstrumentation $instrumentation */
+                $builder = $instrumentation
+                    ->tracer()
+                    ->spanBuilder('publish') // topicName + publish, example foo publish
```

Comment:
Would it be possible to incorporate the span value with the name of the topic and operation as recommended by the semantic convention?